typing/ctype.mli

```diff
@@ -288,20 +288,26 @@ val extract_concrete_typedecl:
 
 val get_new_abstract_name : Env.t -> string -> string
 
-val unify: Env.t -> type_expr -> type_expr -> unit
+val unify_ex: Env.t -> type_expr -> type_expr -> unit
         (* Unify the two types given. Raise [Unify] if not possible. *)
+val unify: Env.t -> type_expr -> type_expr ->
+        (unit, Errortrace.unification_error) Result.t
```

Comment:
nit: prefer `result` over `Result.t`

I would cite that this is consistent with the compiler codebase, but the compiler uses both `result` and `Result.t` haphazardly ://

Reply:
A side question is do we want to define a local allias for the type `('a, Errortrace.unification_error) Result.t` such as `' unification_result` ?